src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1221,11 +1221,87 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _updateSelectionUI();
     }
 
+    static wil::unique_close_clipboard_call _openClipboard(HWND hwnd)
+    {
+        bool success = false;
+
+        // OpenClipboard may fail to acquire the internal lock --> retry.
+        for (DWORD sleep = 10;; sleep *= 2)
+        {
+            if (OpenClipboard(hwnd))
+            {
+                success = true;
+                break;
+            }
+            // 10 iterations
+            if (sleep > 10000)
+            {
+                break;
+            }
+            Sleep(sleep);
+        }
+
+        return wil::unique_close_clipboard_call{ success };
+    }

```

Comment:
This is a copy of the function from `TerminalPage`. My hope is that we resolve the duplicate at a future time and that the linker will fold it away until then.

Reply:
FWIW, clipboard handling was kept outside of Control so that it could be customized by the owning app (e.g. visual studio putting it into local paste history or what-have-you). I am not *against* having copy behavior reside in the control, or in the UI-agnostic inner layers (however: clipboard IO requires an HWND…), but it is a significant departure from that goal.

Reply:
(Alas, the linker will not fold the copies away because they reside in different DLLs!)

Reply:
If we already support such usage of the API I'd be super happy to figure something out to make it work. However, if it was only a plan so far, we may IMHO be putting the cart before the horse: We can easily _add_ clipboard customization at any time if needed, but we can't take it away. Not having it customizable makes the code simpler and more robust - even at the standard 120x9001 buffer size the HTML and RTF contents can easily be >40MB if the extra `HSTRING` conversions are used.